.github/type_check.py

```diff
@@ -1,15 +1,60 @@
+import json
 from pathlib import Path
 import subprocess
+from typing import Any, Dict, List, Union
 
-config = Path(__file__).parent / "pyright-config.json"
+source_dir = Path(__file__).parent
+config = source_dir / "pyright-config.json"
 
-command = ("pyright", "-p", str(config))
-print(" ".join(command))
 
-try:
-    result = subprocess.run(command)
-except FileNotFoundError as e:
-    print(f"{e} - Is pyright installed?")
-    exit(1)
+def pyright() -> int:
+    """ returns process return code """
```

Comment:
Afaik the generic name is exit code or exist status, since it's the integer you pass to exit.
```suggestion
def run_pyright() -> int:
    """ returns process exit code """
```